plugins/source/stripe/resources/services/events/events.go

```diff
@@ -0,0 +1,77 @@
+package events
+
+import (
+	"context"
+
+	"fmt"
+	"strconv"
+
+	"github.com/cloudquery/cloudquery/plugins/source/stripe/client"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+	"github.com/stripe/stripe-go/v74"
+)
+
+func Events() *schema.Table {
+	return &schema.Table{
+		Name:        "stripe_events",
+		Description: `https://stripe.com/docs/api/events`,
+		Transform:   transformers.TransformWithStruct(&stripe.Event{}, client.SharedTransformers(transformers.WithSkipFields("APIResource", "ID"))...),
+		Resolver:    fetchEvents,
+
+		Columns: []schema.Column{
+			{
+				Name:     "id",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("ID"),
+				CreationOptions: schema.ColumnCreationOptions{
+					PrimaryKey: true,
+				},
+			},
+			{
+				Name:     "created",
+				Type:     schema.TypeTimestamp,
+				Resolver: schema.PathResolver("Created"),
+				CreationOptions: schema.ColumnCreationOptions{
+					IncrementalKey: true,
+				},
+			},
+		},
+		IsIncremental: true,
+	}
+}
+
+func fetchEvents(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+	cl := meta.(*client.Client)
+
+	lp := &stripe.EventListParams{}
+
+	const key = "events"
+
+	if cl.Backend != nil {
+		value, err := cl.Backend.Get(ctx, key, cl.ID())
+		if err != nil {
+			return fmt.Errorf("failed to retrieve state from backend: %w", err)
+		}
+		if value != "" {
+			vi, err := strconv.ParseInt(value, 10, 64)
+			if err != nil {
+				return fmt.Errorf("retrieved invalid state value: %q %w", value, err)
+			}
+			lp.Created = &vi
+		}
+	}
+
+	it := cl.Services.Events.List(lp)
+	for it.Next() {
+		data := it.Event()
+		lp.Created = client.MaxInt64(lp.Created, &data.Created)
+		res <- data
```

Comment:
Ideally we should occasionally update the state while the sync is happening as well, in case it fails or stops halfway through

Reply:
I am not sure we can do that... Just because the data was grabbed doesn't mean it was persisted in the datastore...